service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -50,6 +61,67 @@ public DynamicList getHearingDateDynamicList(CaseDetails caseDetails) {
         return hearingDatesDynamic;
     }
 
+    public String createRespondentAssigneeDropdownKey(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (isNotEmpty(caseData.getRespondents1())) {
+            for (int i = 0; i < caseData.getRespondents1().size(); i++) {
+                RespondentParty respondentParty = caseData.getRespondents1().get(i).getValue().getParty();
+
+                String key = String.format("Respondent %d - %s", i + 1, getRespondentFullName(respondentParty));
+                stringBuilder.append(key).append("\n\n");
+            }
+        }
+
+        return stringBuilder.toString().stripTrailing();
+    }
+
+    public String createOtherPartiesAssigneeDropdownKey(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (isNotEmpty(caseData.getOthers())) {
+            for (int i = 0; i < caseData.getOthers().getAllOthers().size(); i++) {
+                Other other = caseData.getOthers().getAllOthers().get(i);
+                String key;
+
+                if (i == 0) {
+                    key = String.format("Person %d - %s", i + 1, defaultIfNull(other.getName(), ""));
```

Comment:
Can person name be null or empty? If so it will render `Person 1 - ` which does not look great. Dash should be added only when name is available.

Reply:
It can, and there is also no safety guards on respondents fullName being empty string. I'd thought about this and asked Moray about placeholders if fullName was empty in both cases. To which I was pointed to a ticket https://tools.hmcts.net/jira/browse/FPLA-962 created just for the validation on this. 

It would be difficult to add validation without partially completing the above ticket.

Reply:
Caught up with Moray, if firstName does not exist it will read 'BLANK - Please complete'